app/validators/valid_text_validator.rb

```diff
@@ -1,9 +1,11 @@
 # frozen_string_literal: true
 
 class ValidTextValidator < ActiveModel::EachValidator
-  ERROR_MESSAGE = 'deve ser um valor válido'
-
   def validate_each(record, attribute, value)
-    record.errors.add(attribute, ERROR_MESSAGE) unless Ratel::Validator.valid?(value.to_s)
+    record.errors.add(attribute, "#{bad_word(value)} é inválido(a)") unless Ratel::Validator.valid?(value.to_s)
```

Comment:
Aqui, você ta fazendo a checagem de erros duas vezes, a primeira no `valid?` e a segunda no `bad_word`.

Que tal algo assim:
```suggestion
    bad_word = Ratel::Validator.invalid_word(value.to_s)
    return unless bad_word
    
    record.errors.add(attribute, "#{bad_word} é inválido(a)")
```